trainings/insights/lesson-plan.md

```diff
@@ -0,0 +1,84 @@
+
+
+# Code Insights Webinar
+
+**Topic:** This training is intended as an initial deep-dive to educate users on Code Insights outside of the 101/102 context. **If at all possible, we recommend having the customer provide a Code Insights use case to you *before* the training, giving you the opportunity to work on that specific insight or something similar;** if not, you can use this as a way to show Code Insights to them and discuss use cases they might find valuable live on the training.
+
+## Intro
+
+Today's training is intended to walk users through our Code Insights product, and show users how to use the feature. By the end of it, we hope that you'll feel comfortable with what Code Insights is and can do, and will be ready to start with your own changes in Sourcegraph.
+
+❗️ If the training is happening in the context of a champion bringing in other team members, have them speak at the top about why they think Code Insights will be useful, and what they envision the team using it for.
+
+## Unit 1: Get familiar with Code Insights
+
+**Learning goals:** After completing this unit, customers will understand what Code Insights are, and can be used to track.
+
+### What is Code Insights?
+
+Code Code Insights reveals high-level information about your codebase, based on both how your code changes over time and its current state. It's a way of visualizing your searches as graphs, so that you can track not just what things are like today, but how that has changed over time. This can be super helpful for tracking things like:
+
+* Migrating from one version of a language or dependency to another
+* Whether folks are adding frequent linter overrides 
+* Tracking how many TODOs are present in your code
+* How adoption of a new API is going
+
+With this info, you can easily make informed decisions about priorities and where to focus your time and energy.
+
+### Running an Insight
+
+To get started, I'm going to generate an insight that tracks TypeScript utilization versus JavaScript utilization in my codebase.
```

Comment:
I'm curious why you chose this as the starting point example? I might have gone with a slightly more specific query to really show off the power at first (this is one of the "popular" use cases but I wonder if another would be a better first example: https://docs.sourcegraph.com/code_insights/references/common_use_cases#popular )

Reply:
Ah, I see your reasons below. These are good reasons, I just wonder if a query with an actual "target" and not just filter strings could be good here. I also think migrations and version tracking are both more common use cases. (That said, maybe this is a moot point – if we're doing our jobs we should be able to demonstrate a more relevant query anyway, as yo noted.)

Reply:
My thought was that this lets the CE touch on the filters (useful to reinforce) and everyone has _some_ JavaScript floating around—it's just the most generic possible example. I'll set a reminder to check in in a month and see if folks have found themselves gathering around another default example, and we can swap if needed.